src/dialects/oracle/query-generator.js

```diff
@@ -0,0 +1,1443 @@
+          const column = paths.shift();
+          str = this.jsonPathExtractionQuery(column, paths);
+        }
+        if (smth.value) {
+          str += util.format(' = %s', this.escape(smth.value));
+        }
+
+        return str;
+      }
+    }
+    if (smth instanceof Utils.Cast) {
+      if (smth.val instanceof Utils.SequelizeMethod) {
+        str = this.handleSequelizeMethod(smth.val, tableName, factory, options, prepend);
+        if (smth.type === 'boolean') {
+          str = `(CASE WHEN ${str}='true' THEN 1 ELSE 0 END)`;
+          return `CAST(${str} AS NUMBER)`;
+        } if (smth.type === 'timestamptz' && /json_value\(/.test(str)) {
+          str = str.slice(0, -1);
+          return `${str} RETURNING TIMESTAMP WITH TIME ZONE)`;
+        }
+      }
+    }
+    return super.handleSequelizeMethod(smth, tableName, factory, options, prepend);
+  }
+
+  _checkValidJsonStatement(stmt) {
+    if (typeof stmt !== 'string') {
+      return false;
+    }
+
+    let currentIndex = 0;
+    let openingBrackets = 0;
+    let closingBrackets = 0;
+    let hasJsonFunction = false;
+    let hasInvalidToken = false;
+
+    while (currentIndex < stmt.length) {
+      const string = stmt.substr(currentIndex);
+      const functionMatches = JSON_FUNCTION_REGEX.exec(string);
+      if (functionMatches) {
+        currentIndex += functionMatches[0].indexOf('(');
+        hasJsonFunction = true;
+        continue;
+      }
+
+      const operatorMatches = JSON_OPERATOR_REGEX.exec(string);
+      if (operatorMatches) {
+        currentIndex += operatorMatches[0].length;
+        hasJsonFunction = true;
+        continue;
+      }
+
+      const tokenMatches = TOKEN_CAPTURE_REGEX.exec(string);
+      if (tokenMatches) {
+        const capturedToken = tokenMatches[1];
+        if (capturedToken === '(') {
+          openingBrackets++;
+        } else if (capturedToken === ')') {
+          closingBrackets++;
+        } else if (capturedToken === ';') {
+          hasInvalidToken = true;
+          break;
+        }
+        currentIndex += tokenMatches[0].length;
+        continue;
+      }
+
+      break;
+    }
+
+    // Check invalid json statement
+    if (hasJsonFunction && (hasInvalidToken || openingBrackets !== closingBrackets)) {
+      throw new Error(`Invalid json statement: ${stmt}`);
+    }
+
+    // return true if the statement has valid json function
+    return hasJsonFunction;
+  }
+
+  jsonPathExtractionQuery(column, path) {
+    let paths = _.toPath(path);
+    const quotedColumn = this.isIdentifierQuoted(column) ? column : this.quoteIdentifier(column);
+
+    paths = paths.map(subPath => {
+      return /\D/.test(subPath) ? Utils.addTicks(subPath, '"') : subPath;
+    });
+
+    const pathStr = this.escape(['$'].concat(paths).join('.').replace(/\.(\d+)(?:(?=\.)|$)/g, (__, digit) => `[${digit}]`));
+
+    return `json_value(${quotedColumn},${pathStr})`;
+  }
+
+  addLimitAndOffset(options, model) {
```

Comment:
mssql does the same thing but it's leading to a few bugs, see https://github.com/sequelize/sequelize/pull/14194 for more information about this

Reply:
Fixed it now.